mission_control/navigator_missions/navigator_missions/entrance_gate.py

```diff
@@ -0,0 +1,463 @@
+        parser.add_argument('-s', '--speed', type=float, default=0.75, choices=[Range(0.0, 1.0)],
+                            help='speed to move when scanning in pass pass mode')
+        parser.add_argument('-k', '--kill', type=str2bool, default=True,
+                            help='true to kill thrusters during scan in two points scan mode')
+        parser.add_argument('-t', '--scantime', type=int, default=10,
+                            help='number of seconds to scan at each point for in two points scan mode')
+        parser.add_argument('-d', '--traversaldist', type=int, default=7,
+                            help='distance from each side of the gates to navigate to when crossing')
+        cls.parser = parser
+
+    @util.cancellableInlineCallbacks
+    def run(self, args):
+        # Parse Arguments
+        pass_scan = args.passscan
+        traversal_distance = args.traversaldist
+
+        # Find the gates
+        self.gate_results = yield self.find_gates()
+        self.gate_centers = self.gate_results[0]
+        self.gates_line = self.gate_results[1]
+        self.gate_totems = self.gate_results[2]
+
+        # If nothing is found, just go through gate 2
+        self.pinger_gate = 1
+
+        if pass_scan:
+            yield self.run_pass_scan(args)
+        else:
+            yield self.run_dual_scan(args)
+
+        self.send_feedback('Gate identified as ' + str(self.pinger_gate + 1))
+
+        # Calculate traversal points
+        traversal_points = yield self.get_perpendicular_points(self.gate_centers[self.pinger_gate], traversal_distance)
+
+        # Go through the gate
+        self.send_feedback('Navigating through gate')
+        yield self.move.set_position(traversal_points[0]).look_at(traversal_points[1]).go()
+        yield self.move.set_position(traversal_points[1]).go()
+        self.send_feedback('Done with start gate!')
+
+    @util.cancellableInlineCallbacks
+    def run_dual_scan(self, args):
+        # Parse Parameters
+        uses_multilateration = args.multilateration
+        scan_dist = args.scandist
+        should_kill = args.kill
+        listen_time = args.scantime
+
+        # Calculate scan points
+        scan_points_0 = yield self.get_perpendicular_points(self.gate_totems[1], scan_dist)
+        scan_points_1 = yield self.get_perpendicular_points(self.gate_totems[2], scan_dist)
+        scan_point_0 = scan_points_0[0]
+        scan_lookat_0 = scan_points_0[1]
+        scan_point_1 = scan_points_1[0]
+        scan_lookat_1 = scan_points_1[1]
+
+        if uses_multilateration:
+            # Reset Scan
+            yield self.hydrophones.disable()
+            yield self.hydrophones.reset()
+        else:
+            # Reset pings
+            self.intersect_vectors = []
+
+        # Execute Scan 1
+
+        self.send_feedback('Navigating to scan start point 1')
+        yield self.move.set_position(scan_point_0).look_at(scan_lookat_0).go()
```

Comment:
This should probably be .look_at().set_position() to minimize travel time. Looking at a point you are already on top of is probably unreliable / random.

Reply:
It's actually looking at a point that's on the other side of the gate, perpedicular through the gate. 'get_perpendicular_points' returns two points, one on the boat's side of the gate, and the other on the opposite side of the gate.
https://github.com/jaxnb/NaviGator/blob/startgate/mission_control/navigator_missions/navigator_missions/entrance_gate.py#L95